### DISCUSSION TITLE
qsv example?

### DISCUSSION THREAD

**Discussion by Tryanks** [asticode/go-astiav#104] [General]
```go
	Device, _ := astiav.CreateHardwareDeviceContext(astiav.HardwareDeviceTypeQSV, "auto", nil)
	Context.SetHardwareDeviceContext(Device)
```
Based on the C version example code, I think only these two lines (error handling ignored) need to be added to enable QSV, am I correct?
If it's convenient for you, could you provide a complete QSV example directly?

**Comment by asticode** [comment]
I'm not familiar enough with QSV to understand fully what `dynamic_set_parameter` is doing and that part is missing in the example. Apart from that, I don't really see the point of adding a specific example for qsv except if the current hardware decode and hardware encode examples don't work properly with qsv. But I'd rather make only one example for hardware and not split it for every device type
